directives/animate/animate.js

```diff
@@ -291,50 +289,32 @@ class AnimationState {
 }
 
 export class Hide extends AsyncDirective {
-	state;
-	constructor(propertyPart) {
-		super(propertyPart);
-		if (!(propertyPart.type === PropertyPart) || propertyPart.committer.name !== 'animate') {
-			throw new Error('animation directives must be used with "animate" property');
-		}
+	constructor(partInfo) {
+		super(partInfo);
+		this.state = new AnimationState(partInfo);
 	}
-	// Do SSR-compatible rendering (arguments are passed from call site)
-	render(part) {
-		// Previous state available on class field
-		if (this.state === undefined) {
-			this.state = new AnimationState(part);
-		}
+	render() {
 		return noChange;
 	}
-	update(part, opts) {
-		/* Any imperative updates to DOM/parts would go here */
+	update(part, [opts]) {
 		opts = opts || {};
 		this.state.hide(opts);
 		return this.render(part);
 	}
 }
 
 export class Show extends AsyncDirective {
-	state;
-	constructor(propertyPart) {
-		super(propertyPart);
-		if (!(propertyPart instanceof PropertyPart) || propertyPart.committer.name !== 'animate') {
-			throw new Error('animation directives must be used with "animate" property');
-		}
+	constructor(partInfo) {
+		super(partInfo);
+		this.state = new AnimationState(partInfo);
 	}
-	// Do SSR-compatible rendering (arguments are passed from call site)
-	render(part) {
-		// Previous state available on class field
-		if (this.state === undefined) {
-			this.state = new AnimationState(part);
-		}
+	render() {
 		return noChange;
 	}
 	update(part, [opts]) {
-		/* Any imperative updates to DOM/parts would go here */
 		opts = opts || {};
 		this.state.show(opts);
-		return this.render(part);
+		return this.render(opts);
```

Comment:
This change seemed to fix things. I don't even know if we need the `render()` at all actually.